bitwarden_license/src/Commercial.Core/AdminConsole/Services/ProviderService.cs

```diff
@@ -299,6 +311,19 @@ public async Task<List<Tuple<ProviderUser, string>>> ConfirmUsersAsync(Guid prov
                     throw new BadRequestException("Invalid user.");
                 }
 
+                if (_featureService.IsEnabled(FeatureFlagKeys.AutomaticConfirmUsers))
+                {
+                    var organizationAutoConfirmPolicyRequirement = await _policyRequirementQuery
+                        .GetAsync<AutomaticUserConfirmationPolicyRequirement>(user.Id);
+
+                    if (organizationAutoConfirmPolicyRequirement
+                        .UserBelongsToOrganizationWithAutomaticUserConfirmationEnabled())
+                    {
+                        result.Add(Tuple.Create(providerUser, new ProviderUsersCannotJoin().Message));
+                        continue;
+                    }
+                }
```

Comment:
This database call is in a loop. We have a bulk interface for `IPolicyRequirementQuery` you can use to get all of them upfront.